From 34fe4e5ab1b9dc50db5562beb0225c1baf12689a Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Thu, 18 Nov 2021 17:06:18 +0800
Subject: [PATCH 48/64] Revert "drm/msm/dp: return correct edid checksum after
 corrupted edid checksum read"

This reverts commit f86bc4a1a401d3691bad41e67f9db0c2ea94da32.
---
 drivers/gpu/drm/msm/dp/dp_panel.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/msm/dp/dp_panel.c b/drivers/gpu/drm/msm/dp/dp_panel.c
index 2768d1d306f0..18cec4fc5e0b 100644
--- a/drivers/gpu/drm/msm/dp/dp_panel.c
+++ b/drivers/gpu/drm/msm/dp/dp_panel.c
@@ -261,7 +261,7 @@ static u8 dp_panel_get_edid_checksum(struct edid *edid)
 {
 	struct edid *last_block;
 	u8 *raw_edid;
-	bool is_edid_corrupt = false;
+	bool is_edid_corrupt;
 
 	if (!edid) {
 		DRM_ERROR("invalid edid input\n");
@@ -293,12 +293,7 @@ void dp_panel_handle_sink_request(struct dp_panel *dp_panel)
 	panel = container_of(dp_panel, struct dp_panel_private, dp_panel);
 
 	if (panel->link->sink_request & DP_TEST_LINK_EDID_READ) {
-		u8 checksum;
-
-		if (dp_panel->edid)
-			checksum = dp_panel_get_edid_checksum(dp_panel->edid);
-		else
-			checksum = dp_panel->connector->real_edid_checksum;
+		u8 checksum = dp_panel_get_edid_checksum(dp_panel->edid);
 
 		dp_link_send_edid_checksum(panel->link, checksum);
 		dp_link_send_test_response(panel->link);
-- 
2.27.0

